internal/templates/port/template.go

```diff
@@ -0,0 +1,39 @@
+package port
+
+import (
+	"fmt"
+
+	"golang.stackrox.io/kube-linter/internal/check"
+	"golang.stackrox.io/kube-linter/internal/diagnostic"
+	"golang.stackrox.io/kube-linter/internal/objectkinds"
+	"golang.stackrox.io/kube-linter/internal/templates"
+	"golang.stackrox.io/kube-linter/internal/templates/port/internal/params"
+	"golang.stackrox.io/kube-linter/internal/templates/util"
+	v1 "k8s.io/api/core/v1"
+)
+
+func init() {
+	templates.Register(check.Template{
+		HumanName:   "ports",
+		Key:         "ports",
+		Description: "Flag containers exposing ports which not recommended as per security best practice",
+		SupportedObjectKinds: check.ObjectKindsDesc{
+			ObjectKinds: []string{objectkinds.DeploymentLike},
+		},
+		Parameters:             params.ParamDescs,
+		ParseAndValidateParams: params.ParseAndValidate,
+		Instantiate: params.WrapInstantiateFunc(func(p params.Params) (check.Func, error) {
+			return util.PerContainerCheck(func(container *v1.Container) []diagnostic.Diagnostic {
+				var results []diagnostic.Diagnostic
+				for _, port := range container.Ports {
+					if int(port.ContainerPort) == p.Port && string(port.Protocol) == p.Protocol {
```

Comment:
for matching protocol, please use `matcher.ForString`

a usage example: https://github.com/stackrox/kube-linter/blob/main/internal/templates/envvar/template.go#L28 :)